packages/providers/src/fuel-graphql-request.ts

```diff
@@ -0,0 +1,154 @@
+/* 
+
+  Replaces `graphql-request` and `graphql-tag` dependencies with a simple
+  `fetch` implementation, reducing the final bundle size by 15.68 kB.
+
+  More:
+    https://github.com/FuelLabs/fuels-ts/discussions/1592#discussioncomment-8003515
+
+*/
+
+import { FuelError } from '@fuel-ts/errors';
+import type { DocumentNode } from 'graphql';
+import { print } from 'graphql';
+
+function isSubscription({ definitions: defs }: DocumentNode) {
+  const opDef = defs.find((op) => op.kind === 'OperationDefinition');
+
+  if (opDef && 'operation' in opDef) {
+    return opDef?.operation === 'subscription';
+  }
+
+  return false;
+}
+
+class FuelGraphqlSubscriber implements AsyncIterator<unknown> {
+  private stream!: ReadableStreamDefaultReader<Uint8Array>;
+  private static textDecoder = new TextDecoder();
+
+  private static parseBytesStream(
+    bytes: Uint8Array | undefined
+  ): Record<string, unknown> | FuelError | undefined {
+    if (bytes === undefined) {
+      return undefined;
+    }
+
+    const text = this.textDecoder.decode(bytes);
+
+    // https://github.com/FuelLabs/fuel-core/blob/e1e631902f762081d2124d9c457ddfe13ac366dc/crates/fuel-core/src/graphql_api/service.rs#L247
+    if (text === 'keep-alive-text') {
+      return undefined;
+    }
+
+    const { data, errors } = JSON.parse(text.split('data:')[1]);
+
+    if (Array.isArray(errors)) {
+      return new FuelError(
+        FuelError.CODES.INVALID_REQUEST,
+        errors.map((err) => err.message).join('\n\n')
+      );
+    }
+
+    return data as Record<string, unknown>;
+  }
+
+  public constructor(
+    private options: {
+      url: string;
+      request: RequestInit;
+      fetchFn: typeof fetch;
+    }
+  ) {}
+
+  private async setStream() {
+    const { url, request, fetchFn } = this.options;
+    const response = await fetchFn(`${url}-sub`, {
+      ...request,
+      headers: {
+        ...request.headers,
+        Accept: 'text/event-stream',
+      },
+    });
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    this.stream = response.body!.getReader();
+  }
+
+  private async readStream(): Promise<IteratorResult<unknown, unknown>> {
+    const { value, done } = await this.stream.read();
```

Comment:
Operations on streams can fail due to unexpected reasons ( network issues ).

I believe we need to handle any potential error here:

```ts
    try {
      const { value, done } = await this.stream.read();

      const parsed = FuelGraphqlSubscriber.parseBytesStream(value);

      if (parsed === undefined && !done) {
        return this.readStream();
      }

      return { value: parsed, done };
    } catch(err) {
      // Handling the error
    }
```

Reply:
The thought process in https://github.com/FuelLabs/fuels-ts/pull/1597#discussion_r1453626098 applies here as well, but with the added caveat that the call to `this.stream.read()` will only happen if the connection was already established. I just tried locally spinning up a node, subscribing to it and then killing it while the subscription connection is still active and I got a `TypeError: fetch failed` error soon after, which means that node throws errors when the underlying stream source is abruptly closed due to a network failure. As stated in the linked comment, I prefer not obfuscating the errors in any way.